### PR TITLE
Unify wallet creation and ensure .address is always present

### PR DESCRIPTION
**Pull request by leomassazza**
Propagate fix applied in [PR-1354](https://github.com/Synthetixio/synthetix/pull/1354/files) to the rest of the modified scripts as part of the web3 removal. Also unifying the way to get the wallet with or without privateKey and ensuring wallet.address is always present.

### CODE REVIEW

File: publish/src/commands/import-fee-periods.js [hunk 1]
```diff
@@ -69,14 +69,14 @@ const importFeePeriods = async ({
...
-	if (useFork) {
+	if (!privateKey) {
 		const account = getUsers({ network, user: 'owner' }).address; // protocolDAO
 		wallet = provider.getSigner(account);
+		wallet.address = await wallet.getAddress();
 	} else {
 		wallet = new ethers.Wallet(privateKey, provider);
 	}
 
-	if (!wallet.address) wallet.address = wallet._address;
```
Comment (leomassazza) [inline publish/src/commands/import-fee-periods.js:79]: a wallet created with new Wallet will always have .address property. It's only needed while creating it as provider signer.

File: publish/src/commands/migrate-binary-option-markets.js [hunk 2]
```diff
@@ -55,7 +55,6 @@ const migrateBinaryOptionMarkets = async ({
 
 	const provider = new ethers.providers.JsonRpcProvider(providerUrl);
 	const wallet = new ethers.Wallet(privateKey, provider);
-	if (!wallet.address) wallet.address = wallet._address;
```
Comment (leomassazza) [inline publish/src/commands/migrate-binary-option-markets.js:58]: not needed here

File: publish/src/commands/nominate.js [hunk 3]
```diff
@@ -84,11 +84,12 @@ const nominate = async ({
 	if (!privateKey) {
 		const account = getUsers({ network, user: 'owner' }).address; // protocolDAO
 		wallet = provider.getSigner(account);
+		wallet.address = await wallet.getAddress();
 	} else {
 		wallet = new ethers.Wallet(privateKey, provider);
 	}
 
-	const signerAddress = await wallet.getAddress();
+	const signerAddress = wallet.address;
```
Comment (leomassazza) [inline publish/src/commands/nominate.js:92]: obtaining the signerAddress from the wallet.address (we can remove this constant at all and refer to wallet.address

File: publish/src/commands/settle.js [hunk 4]
```diff
@@ -71,14 +71,14 @@ const settle = async ({
 	console.log(gray('gasPrice'), yellow(gasPrice));
 	gasPrice = ethers.utils.parseUnits(gasPrice, 'gwei');
 
-	let deployer;
-	if (useFork) {
+	let wallet;
```
Comment (leomassazza) [inline publish/src/commands/settle.js:74]: replaced `deployer` with `wallet` in the sake of uniformity with the rest of the scripts

**Review by ajsantander — APPROVED** [review]
Love the final pattern. Thanks for the clarity @leomassazza

**Pull request merged**